### PR TITLE
Paulo Barata

### PR DESCRIPTION
**Pull request by PBara7a**

### CODE REVIEW

File: src/components/ContactsEdit.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const [formData, setFormData] = useState(null)
+  const navigate = useNavigate()
+  const params = useParams()
+
+  const updateContact = (newContact) => {
+    const newContacts = [...contacts]
+    const contactToUpdateIndex = newContacts.findIndex(el => el.id === newContact.id)
+    newContacts[contactToUpdateIndex] = {...newContact}
+    setContacts([...newContacts])
+  }
```
Comment (Creatorrr1) [inline src/components/ContactsEdit.js:15]: Nice, contact updated using filter by id

File: src/components/ContactsEdit.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    const opts = {
+      method: "PUT",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(formData)
+    }
+    fetch(`http://localhost:4000/contacts/${params.id}`, opts)
+      .then(res => res.json())
+      .then(data => updateContact(data))
+
+    navigate("/")
```
Comment (Creatorrr1) [inline src/components/ContactsEdit.js:43]: this is great, on submit the user is redirected to the homepage

File: src/components/ContactsView.js [hunk 3]
```diff
@@ -35,6 +35,9 @@ function ContactsView() {
         <a href={`https://twitter.com/${contact.twitter}`} target="_blank">
           {contact.twitter}
         </a>}
+        <Link to={`/contacts/${contact.id}/edit`} >
+          <p>Edit</p>
+        </Link>
```
Comment (Creatorrr1) [inline src/components/ContactsView.js:42]: Like how you made an edit link in the contacts view page that redirects user to the edit page

File: package-lock.json [hunk 4]
```diff
@@ -27392,6 +28378,11 @@
...
       }
     },
+    "stylis": {
+      "version": "4.0.13",
+      "resolved": "https://registry.npmjs.org/stylis/-/stylis-4.0.13.tgz",
+      "integrity": "sha512-xGPXiFVl4YED9Jh7Euv2V220mriG9u4B2TA6Ybjc1catrstKD2PpIdU3U0RKpkVBC2EhmL/F0sPCr9vrFTNRag=="
+    },
     "supports-color": {
       "version": "5.5.0",
       "resolved": "https://registry.npmjs.org/supports-color/-/supports-color-5.5.0.tgz",
```
Comment (Alex90Jennings) [inline package-lock.json:28388]: almost 28400 lines of code, why would you do that to us Paolo? :-)

File: src/components/MeetingsList.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+
+function MeetingsList() {
+  const { id } = useParams()
+  const initialData = { about: "", location: "", time: "", contactId: id }
+  const [meetingData, setMeetingData] = useState(initialData)
+  const [meetings, setMeetings] = useState([])
```
Comment (Alex90Jennings) [inline src/components/MeetingsList.js:8]: nice setup of meetings states/initial data

File: src/components/MeetingsList.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+  useEffect(() => {
+    fetch(`http://localhost:4000/meetings?contactId=${id}`)
+      .then(res => res.json())
+      .then(data => setMeetings(data))
+  }, [id])
```
Comment (Alex90Jennings) [inline src/components/MeetingsList.js:14]: what does the [id] do here?

File: src/useFetch.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+
+export const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const controller = new AbortController();
```
Comment (Alex90Jennings) [inline src/useFetch.js:9]: is this for the loading spinner?

File: src/useFetch.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      fetchData();
+    }, 1000);
+
+    return () => {
+      controller.abort();
+    };
+  }, [url]);
+
+  return { data, isPending, error };
+};
```
Comment (Alex90Jennings) [inline src/useFetch.js:44]: loads of functionality here, well organised in plenty of explicitly named components, easy to read.

File: src/components/MeetingsList.js (hunk 6 above)
Comment (PBara7a) [reply src/components/MeetingsList.js:14]: The fetch url depends on 'id' so I added 'id' to the array of things to track, so the function runs everytime 'id' changes.

File: src/useFetch.js (hunk 7 above)
Comment (PBara7a) [reply src/useFetch.js:9]: That whole file is the custom hook that Alex shared, I made a small change to it to purposely delay the fetching of data and make it easier to see the loading spinner, but the hook itself is not related to the spinner.

**Pull request closed without merging**